libs/components/forms/src/lib/modules/input-box/input-box-host.service.ts

```diff
@@ -40,4 +40,14 @@ export class SkyInputBoxHostService {
 
     this.#host.populate(args);
   }
+
+  public setHintText(hintText: string | undefined): void {
```

Comment:
Do we need to support `undefined` values? I'd rather be deliberate in our API so we don't need to worry about branching logic. The consumer of the input box service would determine if the hint text is defined before asking the input box to set it for them.

Reply:
We have to support `undefined` for clearing the hint text. In theory - I guess that could be done via passing in an empty string - but I personally don't like forcing consumers to do that and it goes against what we do in other places where we allow `undefined` to clear a value like this. 

For datepicker we don't have a case where the host hint text is set and then is cleared - but I could definitely see that happening in the future.

Reply:
I hadn't thought about _unsetting_ the hint text. Good call.